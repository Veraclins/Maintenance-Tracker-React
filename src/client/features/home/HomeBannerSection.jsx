import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

const HomeBannerSection = ({ user, isLoggedIn, isAdmin }) => (
  <div className="content_banner">
    <div className="content_banner_inner">
      <div className="banner_top_bg" />
      <div className="banner_left_bg" />
      <div className="container">
        <div className="inner_layout wow fadeInUp">
          {isLoggedIn ? (
            <h1 className="home_welcome">
              Welcome back{' '}
              <strong>
                {user.firstName} {user.lastName}
              </strong>
            </h1>
          ) : (
            false
          )}
          <h1>
            <strong>This is VeraTech Solutions Hub.</strong>
          </h1>
          <h1>
            The Home of World Class Maintenance and Repairs for your{' '}
            <i>Modern Devices.</i>
          </h1>
          {!isLoggedIn ? (
            <h5>
              VeraTech Solutions Hub gives you the peace of mind you need. We
              care for your devices so you won{"'"}t lose sleep. Whatever the need
              of your device, we got you covered.
            </h5>
          ) : (
            false
          )}
          {isLoggedIn ? (
            <React.Fragment>
              <div className="buttons_banner">
                {' '}
                <Link to="/requests/create" className="buttons button_default">
                  CREATE A NEW REQUEST
                </Link>{' '}
                <Link to="/dashboard" className="buttons button_blank">
                  GO TO YOUR DASHBOARD
                </Link>
              </div>
              {isAdmin ? (
                <div className="buttons_banner">
                  {' '}
                  <Link to="/admin" className="buttons button_default">
                    GO TO ADMIN PAGE
                  </Link>
                </div>
              ) : (
                false
              )}
            </React.Fragment>
          ) : (
            <div className="buttons_banner">
              {' '}
              <Link to="/signup" className="buttons button_default">
                CREATE A NEW ACCOUNT
              </Link>{' '}
              <Link to="/login" className="buttons button_blank">
                LOGIN TO YOUR ACCOUNT
              </Link>
            </div>
          )}
        </div>
      </div>{' '}
      <span className="banner_big_title">RELIABILITY</span>
      <div
        className="vertical_anchor left_vertical wow fadeInLeft"
        data-wow-delay="0.4s"
      >
        {' '}
        <Link to="/signup">CREATE_A_NEW_ACCOUNT</Link>
      </div>
      <div
        className="vertical_anchor right_vertical wow fadeInRight"
        data-wow-delay="0.4s"
      >
        {' '}
        <Link to="/login">LOGIN_TO_YOUR_ACCOUNT</Link>
      </div>
    </div>
  </div>
);

HomeBannerSection.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  user: PropTypes.shape({}).isRequired,
};

export default HomeBannerSection;
