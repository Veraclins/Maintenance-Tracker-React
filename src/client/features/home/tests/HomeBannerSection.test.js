import React from 'react';
import { shallow } from 'enzyme';

import HomeBannerSection from '../HomeBannerSection';

const user = {
  firstName: 'Clinton',
  lastName: 'Agada',
};

describe('Test the HomeBannerSection component', () => {
  it('renders without crashing for a guest', () => {
    const wrapper = shallow(<HomeBannerSection user={user} isLoggedIn={false} isAdmin={false} />);
    expect(wrapper.find('div')).toHaveLength(9);
    expect(wrapper.find('h1.home_welcome')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with the right view for a logged in user', () => {
    const wrapper = shallow(<HomeBannerSection user={user} isLoggedIn isAdmin={false} />);
    expect(wrapper.find('h1.home_welcome')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
});
