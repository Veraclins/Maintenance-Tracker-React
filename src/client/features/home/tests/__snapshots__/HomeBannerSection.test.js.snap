// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the HomeBannerSection component renders with the right view for a logged in user 1`] = `
<div
  className="content_banner"
>
  <div
    className="content_banner_inner"
  >
    <div
      className="banner_top_bg"
    />
    <div
      className="banner_left_bg"
    />
    <div
      className="container"
    >
      <div
        className="inner_layout wow fadeInUp"
      >
        <h1
          className="home_welcome"
        >
          Welcome back
           
          <strong>
            Clinton
             
            Agada
          </strong>
        </h1>
        <h1>
          <strong>
            This is VeraTech Solutions Hub.
          </strong>
        </h1>
        <h1>
          The Home of World Class Maintenance and Repairs for your
           
          <i>
            Modern Devices.
          </i>
        </h1>
        <div
          className="buttons_banner"
        >
           
          <Link
            className="buttons button_default"
            replace={false}
            to="/requests/create"
          >
            CREATE A NEW REQUEST
          </Link>
           
          <Link
            className="buttons button_blank"
            replace={false}
            to="/dashboard"
          >
            GO TO YOUR DASHBOARD
          </Link>
        </div>
      </div>
    </div>
     
    <span
      className="banner_big_title"
    >
      RELIABILITY
    </span>
    <div
      className="vertical_anchor left_vertical wow fadeInLeft"
      data-wow-delay="0.4s"
    >
       
      <Link
        replace={false}
        to="/signup"
      >
        CREATE_A_NEW_ACCOUNT
      </Link>
    </div>
    <div
      className="vertical_anchor right_vertical wow fadeInRight"
      data-wow-delay="0.4s"
    >
       
      <Link
        replace={false}
        to="/login"
      >
        LOGIN_TO_YOUR_ACCOUNT
      </Link>
    </div>
  </div>
</div>
`;

exports[`Test the HomeBannerSection component renders without crashing for a guest 1`] = `
<div
  className="content_banner"
>
  <div
    className="content_banner_inner"
  >
    <div
      className="banner_top_bg"
    />
    <div
      className="banner_left_bg"
    />
    <div
      className="container"
    >
      <div
        className="inner_layout wow fadeInUp"
      >
        <h1>
          <strong>
            This is VeraTech Solutions Hub.
          </strong>
        </h1>
        <h1>
          The Home of World Class Maintenance and Repairs for your
           
          <i>
            Modern Devices.
          </i>
        </h1>
        <h5>
          VeraTech Solutions Hub gives you the peace of mind you need. We care for your devices so you won
          '
          t lose sleep. Whatever the need of your device, we got you covered.
        </h5>
        <div
          className="buttons_banner"
        >
           
          <Link
            className="buttons button_default"
            replace={false}
            to="/signup"
          >
            CREATE A NEW ACCOUNT
          </Link>
           
          <Link
            className="buttons button_blank"
            replace={false}
            to="/login"
          >
            LOGIN TO YOUR ACCOUNT
          </Link>
        </div>
      </div>
    </div>
     
    <span
      className="banner_big_title"
    >
      RELIABILITY
    </span>
    <div
      className="vertical_anchor left_vertical wow fadeInLeft"
      data-wow-delay="0.4s"
    >
       
      <Link
        replace={false}
        to="/signup"
      >
        CREATE_A_NEW_ACCOUNT
      </Link>
    </div>
    <div
      className="vertical_anchor right_vertical wow fadeInRight"
      data-wow-delay="0.4s"
    >
       
      <Link
        replace={false}
        to="/login"
      >
        LOGIN_TO_YOUR_ACCOUNT
      </Link>
    </div>
  </div>
</div>
`;
